index_search_test.go

```diff
@@ -100,7 +100,7 @@ func TestIndex_Search(t *testing.T) {
 			want: &SearchResponse{
 				Hits: []interface{}{
 					map[string]interface{}{
-						"book_id": float64(1), "title": "Alice In Wonderland",
+						"book_id": float64(123), "title": "Pride and Prejudice",
```

Comment:
Can you give some idea about this `float64(...)`? Why do you need to convert it to float?

Reply:
This is the default behavior of the deserializer in golang you can read in the [documentation](https://pkg.go.dev/encoding/json#Unmarshal)

> To unmarshal JSON into an interface value, Unmarshal stores one of these in the interface value:
bool, for JSON booleans
float64, for JSON numbers
string, for JSON strings
[]interface{}, for JSON arrays
map[string]interface{}, for JSON objects
nil for JSON null

And by default, every number declare are `int` so in:
```
	map[string]interface{}{
		"book_id": 1,
	}
```
`book_id` will be an int that is why I convert it. I didn't find a better way to do it.